### ISSUE TITLE
Nearly all tests fail on Windows

### ISSUE THREAD

**Issue by zoffixznet** [zostay/perl6-IO-Glob#7]
This is on Windows 7 with Rakudo 2017.07:

```
===> Searching for missing dependencies: Test::META
===> Updated cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosyste
ms/master/cpan.json
===> Updated p6c mirror: http://ecosystem-api.p6c.org/projects.json
===> Searching for missing dependencies: META6
===> Searching for missing dependencies: JSON::Class
===> Searching for missing dependencies: JSON::Marshal, JSON::Unmarshal
===> Searching for missing dependencies: JSON::Name
===> Testing: JSON::Name:ver<0.0.2>:auth<github:jonathanstowe>
===> Testing [OK] for JSON::Name:ver<0.0.2>:auth<github:jonathanstowe>
===> Testing: JSON::Marshal:ver<0.0.14>:auth<github:jonathanstowe>
===> Testing [OK] for JSON::Marshal:ver<0.0.14>:auth<github:jonathanstowe>
===> Testing: JSON::Unmarshal:ver<0.08>
===> Testing [OK] for JSON::Unmarshal:ver<0.08>
===> Testing: JSON::Class:ver<0.0.10>:auth<github:jonathanstowe>
===> Testing [OK] for JSON::Class:ver<0.0.10>:auth<github:jonathanstowe>
===> Testing: META6:ver<0.0.19>:auth<github:jonathanstowe>
===> Testing [OK] for META6:ver<0.0.19>:auth<github:jonathanstowe>
===> Testing: Test::META:ver<0.0.13>:auth<github:jonathanstowe>
# the following may make some diagnostics from the module itself
===> Testing [OK] for Test::META:ver<0.0.13>:auth<github:jonathanstowe>
===> Testing: IO::Glob:ver<0.3>:auth<github:zostay>

# Failed test at t\dir.t line 10
# expected: '2'
#      got: '0'

# Failed test at t\dir.t line 11
# expected: 't/fixtures/bar.md'
#      got: (Any)

# Failed test at t\dir.t line 12
# expected: 't/fixtures/foo.md'
#      got: (Any)

# Failed test at t\dir.t line 17
# expected: '2'
#      got: '0'

# Failed test at t\dir.t line 18
# expected: 't/fixtures/foo.md'
#      got: (Any)

# Failed test at t\dir.t line 19
# expected: 't/fixtures/foo.txt'
#      got: (Any)

# Failed test at t\dir.t line 25
# expected: 't/fixtures/.'
#      got: 't/fixtures\.'

# Failed test at t\dir.t line 26
# expected: 't/fixtures/..'
#      got: 't/fixtures\..'

# Failed test at t\dir.t line 27
# expected: 't/fixtures/bar.md'
#      got: 't/fixtures\bar.md'

# Failed test at t\dir.t line 28
# expected: 't/fixtures/bar.txt'
#      got: 't/fixtures\bar.txt'

# Failed test at t\dir.t line 29
# expected: 't/fixtures/foo.md'
#      got: 't/fixtures\foo.md'

# Failed test at t\dir.t line 30
# expected: 't/fixtures/foo.txt'
#      got: 't/fixtures\foo.txt'

# Failed test at t\iterator.t line 10
# expected: '2'
#      got: '0'

# Failed test at t\iterator.t line 11
# expected: 't/fixtures/bar.md'
#      got: (Any)

# Failed test at t\iterator.t line 12
# expected: 't/fixtures/foo.md'
#      got: (Any)

# Failed test at t\path-dir.t line 11
# expected: 't/fixtures/bar.md'
#      got: 't/fixtures\bar.md'

# Failed test at t\path-dir.t line 12
# expected: 't/fixtures/foo.md'
#      got: 't/fixtures\foo.md'

# Failed test at t\path-dir.t line 18
# expected: 't/fixtures/foo.md'
#      got: 't/fixtures\foo.md'

# Failed test at t\path-dir.t line 19
# expected: 't/fixtures/foo.txt'
#      got: 't/fixtures\foo.txt'

# Failed test at t\path-dir.t line 25
# expected: 't/fixtures/.'
#      got: 't/fixtures\.'

# Failed test at t\path-dir.t line 26
# expected: 't/fixtures/..'
#      got: 't/fixtures\..'

# Failed test at t\path-dir.t line 27
# expected: 't/fixtures/bar.md'
#      got: 't/fixtures\bar.md'

# Failed test at t\path-dir.t line 28
# expected: 't/fixtures/bar.txt'
#      got: 't/fixtures\bar.txt'

# Failed test at t\path-dir.t line 29
# expected: 't/fixtures/foo.md'
#      got: 't/fixtures\foo.md'

# Failed test at t\path-dir.t line 30
# expected: 't/fixtures/foo.txt'
#      got: 't/fixtures\foo.txt'
# Looks like you failed 10 tests of 13
===> Testing [FAIL]: IO::Glob:ver<0.3>:auth<github:zostay>
Aborting due to test failure: IO::Glob:ver<0.3>:auth<github:zostay> (use --force
-test to override)
  in code  at C:\rakudo\share\perl6\site\sources\0061125F96D1E60B6E3B0001A4CB4D4
188DF4869 (Zef::Client) line 375
  in method test at C:\rakudo\share\perl6\site\sources\0061125F96D1E60B6E3B0001A
4CB4D4188DF4869 (Zef::Client) line 353
  in code  at C:\rakudo\share\perl6\site\sources\0061125F96D1E60B6E3B0001A4CB4D4
188DF4869 (Zef::Client) line 530
  in sub  at C:\rakudo\share\perl6\site\sources\0061125F96D1E60B6E3B0001A4CB4D41
88DF4869 (Zef::Client) line 527
  in method install at C:\rakudo\share\perl6\site\sources\0061125F96D1E60B6E3B00
01A4CB4D4188DF4869 (Zef::Client) line 633
  in sub MAIN at C:\rakudo\share\perl6\site\sources\C360CEB3AF792C33F65155FA82FD
B7AB4467EC82 (Zef::CLI) line 152
  in block <unit> at C:\rakudo\share\perl6\site\resources\94EA132075BAD926331B1E
EE92A28712C70F6F78 line 3
  in sub MAIN at C:\rakudo\share\perl6\site\bin\zef line 2
  in block <unit> at C:\rakudo\share\perl6\site\bin\zef line 2
```

**Comment by zostay** [comment]
Thanks for the report. I haven't run anything on Windows in a long time. Maybe I'll see about getting an EC2 sandbox setup.

**Comment by zoffixznet** [comment]
FWIW there's https://www.appveyor.com/ that's like Travis but for Windows. It's free

**Comment by khalidelboray** [comment]
any updates ?

**Comment by zostay** [comment]
I setup `.appveyor.yml` for this to see what fails and made a rudimentary adjustment to the new `t/absolute.t` test that revealed a new bug. One step forward, one step back.

**Comment by zostay** [comment]
The following problems with tests exist. These are mostly failing because the test assumes the directory separator is `/` when it could be `\`. 

- [ ] `t/dir.t`
- [ ] `t/double-star.t` 
- [ ] `t/iterator.t`
- [ ] `t/path-dir.t`

**Comment by zostay** [comment]
Fixed in e47d6979475c010705bc27d6ac164cbb8f4c1c5d
